src/solver/surfaceIntegrations.F90

```diff
@@ -595,6 +675,32 @@ subroutine wallIntegrationFace(localValues, mm)
           Mv(2) = Mv(2) + my * blk
           Mv(3) = Mv(3) + mz * blk
 
+          ! the force integral for the center of pressure computation.
+          ! We need the cell centers wrt origin
+          xc = fourth*(xx(i,j,  1) + xx(i+1,j,  1) &
```

Comment:
I would changes these to `xco`, `yco`, `zco` as you call them in `flowIntegrationFace`. This helps to distinguish them from distances from the reference point.